packages/scope-manager/src/definition/DefinitionBase.ts

```diff
@@ -5,10 +5,15 @@ import type { DefinitionType } from './DefinitionType';
 import { createIdGenerator } from '../ID';
 
 const generator = createIdGenerator();
+/**
+ * Helper type to exclude Program from valid definition nodes
+ * Program is the root node and doesn't have a parent property
+ */
+export type NodeWithParent = Exclude<TSESTree.Node, TSESTree.Program>;
```

Comment:
🤔 This type doesn't really belong to the scope-manager. Any consumer of the TSESTree nodes is likely to want it, such as the `collectUnusedVariables` change already in this PR. I this this should really be `TSESTree.NodeWithParent`.

Reply:
I got it, will do